processing/src/main/java/org/apache/druid/segment/QueryableIndex.java

```diff
@@ -35,7 +38,7 @@
  * @see QueryableIndexStorageAdapter for query path adapter
  * @see QueryableIndexIndexableAdapter for indexing path adapter
  */
-public interface QueryableIndex extends ColumnSelector, Closeable
+public interface QueryableIndex extends Closeable
```

Comment:
It was not considered.  It might work and might not.  I don't remember places that were creating a cache just to throw it away, IIRC, they were all created in places that could actually have resources that need cleanup, so I think it's fine as is.  If not, I'd put that cleanup in scope for another PR.